gpii/node_modules/lifecycleManager/src/LifecycleManager.js

```diff
@@ -1216,14 +1301,54 @@ var gpii = fluid.registerNamespace("gpii");
                     gpiiKey: undefined,
                     actionResults: {},
                     originalSettings: {},
-                    appliedSolutions: {}
+                    currentSettings: {},
+                    currentPreferences: {}
                 }
             });
             transaction.commit();
         });
         return restorePromise;
     };
 
+     /**
+      * Read setting values of one or more solutions.
+      * @param {Object} that - An instance of gpii.lifecycleManager.
+      * @param {Object} solutions - Solutions with setting to be read from the system.
+      * @return {Promise} A promise whose resolved value is the result of setting values.
+      */
+    gpii.lifecycleManager.processRead = function (that, solutions) {
+        var handlerGetPromises = [];
+        fluid.each(solutions, function (solution, solutionID) {
+            var newSolution = gpii.lifecycleManager.transformSolutionSettings(solution, gpii.settingsHandlers.changesToSettings);
+            fluid.each(newSolution.settingsHandlers, function (handlerSpec) {
+                //if applicationSettings is unspecified, dont attempt to get settings
+                if (!handlerSpec.supportedSettings) {
+                    return;
+                }
+                handlerSpec.settings = handlerSpec.supportedSettings;
+
+                // TODO GPII-228 This is temporary debugging code while we work through
```

Comment:
Is this still temporary debugging code? This JIRA appears to be "Implement snapshotter", has some material leaked into this pull by mistake somehow?

Reply:
This read() function is copied from [@sgithens's snapshotter pull request](https://github.com/GPII/universal/pull/616/files#diff-ccff73df5921887c4a3f5dbbb5182496R162-R184). :) This comment is inherited from the copy. I've adjusted the wording to fit it in this context.